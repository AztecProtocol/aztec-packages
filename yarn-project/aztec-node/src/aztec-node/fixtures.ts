import {
  AccumulatedData,
  AffineElement,
  AggregationObject,
  ConstantData,
  ContractDeploymentData,
  EMITTED_EVENTS_LENGTH,
  EthAddress as CircuitEthAddress,
  Fq,
  Fr,
  FunctionData,
  KERNEL_L1_MSG_STACK_LENGTH,
  KERNEL_NEW_COMMITMENTS_LENGTH,
  KERNEL_NEW_CONTRACTS_LENGTH,
  KERNEL_NEW_NULLIFIERS_LENGTH,
  KERNEL_OPTIONALLY_REVEALED_DATA_LENGTH,
  KERNEL_PRIVATE_CALL_STACK_LENGTH,
  KERNEL_PUBLIC_CALL_STACK_LENGTH,
  NewContractData,
  OldTreeRoots,
  OptionallyRevealedData,
  PrivateKernelPublicInputs,
  TxContext,
  UInt8Vector,
} from '@aztec/circuits.js';
import { EthereumRpc } from '@aztec/ethereum.js/eth_rpc';
import { WalletProvider } from '@aztec/ethereum.js/provider';
import { AztecAddress, randomBytes } from '@aztec/foundation';
import { Rollup, Yeeter } from '@aztec/l1-contracts';
import { Tx } from '@aztec/tx';

// REFACTOR: Move deployment logic to l1-contracts package, and refactor it out of other integration tests (archiver, sequencer)
export const deployRollupContract = async (provider: WalletProvider, ethRpc: EthereumRpc) => {
  const deployAccount = provider.getAccount(0);
  const contract = new Rollup(ethRpc, undefined, { from: deployAccount, gas: 1000000 });
  await contract.deploy().send().getReceipt();
  return contract.address;
};

export const deployYeeterContract = async (provider: WalletProvider, ethRpc: EthereumRpc) => {
  const deployAccount = provider.getAccount(0);
  const contract = new Yeeter(ethRpc, undefined, { from: deployAccount, gas: 1000000 });
  await contract.deploy().send().getReceipt();
  return contract.address;
};

export const createProvider = (host: string, mnemonic: string, accounts: number) => {
  const walletProvider = WalletProvider.fromHost(host);
  walletProvider.addAccountsFromMnemonic(mnemonic, accounts);
  return walletProvider;
};

// REFACTOR: Use @aztec/circuit.js/factories where possible
export const createCircuitEthAddress = () => {
  return new CircuitEthAddress(randomBytes(20));
};

export const createRandomCommitments = (num: number) => {
  return Array(num)
    .fill(0)
    .map(() => new Fr(randomBytes(32)));
};

export const createOptionallyRetrievedData = () => {
  const func = new FunctionData(0, true, true);
  return new OptionallyRevealedData(
    new Fr(0),
    func,
    createRandomCommitments(EMITTED_EVENTS_LENGTH),
    new Fr(0),
    createCircuitEthAddress(),
    true,
    true,
    true,
    true,
  );
};

export const createOptionallyRetrievedDatas = (num: number) => {
  return Array(num)
    .fill(0)
    .map(() => createOptionallyRetrievedData());
};

export const createNewContractData = () => {
  return new NewContractData(AztecAddress.random(), createCircuitEthAddress(), new Fr(randomBytes(32)));
};

export const createNewContractDatas = (num: number) => {
  return Array(num)
    .fill(0)
    .map(() => createNewContractData());
};

export const createTx = () => {
  const oldTreeRoots = new OldTreeRoots(new Fr(0), new Fr(0), new Fr(0), new Fr(0));
  const contractDeploymentData = new ContractDeploymentData(
    new Fr(randomBytes(32)),
    new Fr(randomBytes(32)),
    new Fr(randomBytes(32)),
    createCircuitEthAddress(),
  );
  const txContext = new TxContext(false, false, true, contractDeploymentData);
  const constantData = new ConstantData(oldTreeRoots, txContext);
  const aggregationObject = new AggregationObject(
    new AffineElement(new Fq(0), new Fq(0)),
    new AffineElement(new Fq(0), new Fq(0)),
    [],
    [],
    false,
  );
  const accumulatedData = new AccumulatedData(
    aggregationObject,
    new Fr(0),
    createRandomCommitments(KERNEL_NEW_COMMITMENTS_LENGTH),
    createRandomCommitments(KERNEL_NEW_NULLIFIERS_LENGTH),
    createRandomCommitments(KERNEL_PRIVATE_CALL_STACK_LENGTH),
    createRandomCommitments(KERNEL_PUBLIC_CALL_STACK_LENGTH),
    createRandomCommitments(KERNEL_L1_MSG_STACK_LENGTH),
    createNewContractDatas(KERNEL_NEW_CONTRACTS_LENGTH),
    createOptionallyRetrievedDatas(KERNEL_OPTIONALLY_REVEALED_DATA_LENGTH),
  );
  const kernelInputs = new PrivateKernelPublicInputs(accumulatedData, constantData, true);
  return new Tx(kernelInputs, new UInt8Vector(Buffer.alloc(0)));
};
