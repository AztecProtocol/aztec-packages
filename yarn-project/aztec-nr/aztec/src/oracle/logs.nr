use crate::types::point::Point;
use crate::constants_gen::NUM_FIELDS_PER_SHA256;

// TODO: Should take encrypted data.
#[oracle(emitEncryptedLog)]
fn emit_encrypted_log_oracle<N>(_contract_address: Field, _storage_slot: Field, _encryption_pub_key: Point, _preimage: [Field; N]) -> Field {}

unconstrained pub fn emit_encrypted_log<N>(contract_address: Field, storage_slot: Field, encryption_pub_key: Point, preimage: [Field; N]) -> [Field; NUM_FIELDS_PER_SHA256] {
    [emit_encrypted_log_oracle(contract_address, storage_slot, encryption_pub_key, preimage), 0]
}

#[oracle(emitUnencryptedLog)]
fn emit_unencrypted_log_oracle<T>(_contract_address: Field, _event_selector: Field, _message: T) -> Field {}

unconstrained pub fn emit_unencrypted_log<T>(contract_address: Field, event_selector: Field, message: T) -> [Field; NUM_FIELDS_PER_SHA256] {
    // https://github.com/AztecProtocol/aztec-packages/issues/885
    [emit_unencrypted_log_oracle(contract_address, event_selector, message), 0]
}
