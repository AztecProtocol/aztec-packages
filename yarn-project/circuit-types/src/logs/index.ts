export * from './event_metadata.js';
export * from './get_logs_response.js';
export * from './l2_logs_source.js';
export * from './log_id.js';
export * from './log_filter.js';
export * from './l1_payload/index.js';
export * from './extended_public_log.js';
export * from './extended_contract_class_log.js';
