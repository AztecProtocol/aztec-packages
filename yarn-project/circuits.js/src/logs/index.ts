export * from './log_with_tx_data.js';
export * from './indexed_tagging_secret.js';
export * from './public_log.js';
export * from './private_log.js';
