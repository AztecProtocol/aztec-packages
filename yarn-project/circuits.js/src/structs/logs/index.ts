export * from './log_with_tx_data.js';
