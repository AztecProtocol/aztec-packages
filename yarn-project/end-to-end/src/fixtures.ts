export const MNEMONIC = 'test test test test test test test test test test test junk';
