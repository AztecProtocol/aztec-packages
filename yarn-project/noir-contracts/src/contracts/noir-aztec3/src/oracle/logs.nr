use crate::types::point::Point;
use crate::notes::value_note::Note;

// TODO: Make independent of a specific note implementation.
#[oracle(emitEncryptedLog)]
fn emit_encrypted_log_oracle(
    _contract_address: Field,
    _storage_slot: Field,
    _owner: Point,
    _note: Note,
)  -> Field {}

unconstrained fn emit_encrypted_log(
    contract_address: Field,
    storage_slot: Field,
    owner: Point,
    note: Note,
) -> Field {
    emit_encrypted_log_oracle(
        contract_address, 
        storage_slot,
        owner,
        note,
    )
}