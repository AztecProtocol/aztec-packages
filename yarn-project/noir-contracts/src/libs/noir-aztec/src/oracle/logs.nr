use crate::types::point::Point;
use crate::constants_gen::NUM_FIELDS_PER_SHA256;

// TODO: Should take encrypted data.
#[oracle(emitEncryptedLog)]
fn emit_encrypted_log_oracle<N>(
    _contract_address: Field,
    _storage_slot: Field,
    _owner: Point,
    _preimage: [Field; N],
)  -> Field {}

unconstrained fn emit_encrypted_log<N>(
    contract_address: Field,
    storage_slot: Field,
    owner: Point,
    preimage: [Field; N],
) -> [Field; NUM_FIELDS_PER_SHA256] {
    [emit_encrypted_log_oracle(
        contract_address,
        storage_slot,
        owner,
        preimage,
    ), 0]
}

#[oracle(emitUnencryptedLog)]
fn emit_unencrypted_log_oracle<T>(_message: T)  -> Field {}

unconstrained fn emit_unencrypted_log<T>(message: T) -> [Field; NUM_FIELDS_PER_SHA256] {
    // https://github.com/AztecProtocol/aztec-packages/issues/885
    [emit_unencrypted_log_oracle(message), 0]
}