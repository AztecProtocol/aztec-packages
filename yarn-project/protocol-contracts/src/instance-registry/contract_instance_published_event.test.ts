import { PrivateLog } from '@aztec/stdlib/logs';

import { getSampleContractInstancePublishedEventPayload } from '../tests/fixtures.js';
import { ContractInstancePublishedEvent } from './contract_instance_published_event.js';

describe('ContractInstancePublishedEvent', () => {
  it('parses an event as emitted by the ClassInstanceRegistry', () => {
    const data = getSampleContractInstancePublishedEventPayload();
    const log = PrivateLog.fromBuffer(data);
    expect(ContractInstancePublishedEvent.isContractInstancePublishedEvent(log)).toBe(true);

    const event = ContractInstancePublishedEvent.fromLog(log);

    // See ./__snapshots__/README.md for how to update the snapshot.
    expect(event).toMatchSnapshot();
  });
});
