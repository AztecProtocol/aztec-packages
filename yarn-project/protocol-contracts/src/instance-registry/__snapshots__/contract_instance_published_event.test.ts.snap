// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ContractInstancePublishedEvent parses an event as emitted by the ClassInstanceRegistry 1`] = `
ContractInstancePublishedEvent {
  "address": "0x27ab5678efa16cd36014d6f92d72d89fb2496e4394d8a75ca82674137cd9c1d4",
  "contractClassId": "0x0a8ad282f9ed12cd5f61b8b0224b526b5950ebac30e6660189cb34418b6a3c66",
  "deployer": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "initializationHash": "0x1955df48942dba07aaf78cd693a8322f190638e1adf0264db9374f40d8e4d14f",
  "publicKeys": PublicKeys {
    "masterIncomingViewingPublicKey": "0x2e3836a1ef93615f986841eb5dcd2314bf35ee736120e6645fe99b406e096b8f1f53dee6f3b0b40f17a76a19b7d3e0e3915c209db5cda640173b09307c79df49",
    "masterNullifierPublicKey": "0x008519fee80aaf7006d9b95d804cf35ee32a0d118bfb1387e9aabec1921d989606676bdeefbdd532b3e4ac5b0df708b26045322ac6e8619e5b8130ce00a5d872",
    "masterOutgoingViewingPublicKey": "0x233da67dec0067e2ebec0e0f0f483bd0996cfc79f8581e47ab44be989696c2ea22c3e09c642a449b3b095a945f5e4d800ce6899a607786278256183a0cd2afa2",
    "masterTaggingPublicKey": "0x21239edc1ed21cb3abd14e9c5de3b4a00417a34716cb48c38d3884c2fc5782bd19e707b85bdc938999562ae3939c43c71100dc8ddec2dc18e998de5f3b32723c",
  },
  "salt": "0x00f4851dcaed949c8224bd411cee58a130b29295cf0f94293d62e876f1b36cc9",
  "version": 1,
}
`;
